R/get_hospdata.R

```diff
@@ -0,0 +1,159 @@
+#'  by \code{\link{get_hospdata_long}}.
+#' @param preferred_source character or character vector, either "OWID",
+#'  "ECDC", or c("OWID", "ECDC") for both). Defaults to keeping both sources.
+#'
+#' @return
+#' A data frame with n rows and up to 13 columns:
+#'
+#' \describe{
+#'       \item{id}{character, ISO 3166-1 alpha-3 country code}
+#'       \item{source}{character, ECDC or OWID, see details for more information.}
+#'       \item{date}{date of observation}
+#'       \item{daily_icu_occupancy}{OWID & ECDC: Number of COVID-19 patients in ICU on a given day}
+#'       \item{daily_icu_occupancy_per_million}{OWID: daily_icu_occupancy per million people}
+#'       \item{daily_hospital_occupancy}{OWID & ECDC: Number of COVID-19 patients in hospital on a given day}
+#'       \item{daily_hospital_occupancy_per_million}{OWID: daily_hospital_occupancy per million people}
+#'       \item{weekly_new_hospital_admissions}{OWID & ECDC: Number of COVID-19 patients newly admitted
+#'             to hospitals in a given week}
+#'       \item{weekly_new_hospital_admissions_per_100k}{ECDC: weekly_new_hospital_admissions from ECDC per 100,000 persons}
+#'       \item{weekly_new_hospital_admissions_per_million}{OWID: weekly_new_hospital_admissions from OWID per million people}
+#'       \item{weekly_new_icu_admissions}{OWID & ECDC: Number of COVID-19 patients newly admitted to ICU in a given week}
+#'       \item{weekly_new_icu_admissions_per_million}{OWID: Weekly_new_icu_admissions from OWID per million people}
+#'       \item{weekly_new_icu_admissions_per_100k}{ECDC: weekly_new_icu_admissions from ECDC per 100,000 persons}
+#' }
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#'  hospdata_long <- get_hospdata_long()
+#'  hospdata_wide <- get_hospdata_wide(hospdata_long, preferred_source = "OWID")
+#' }
+#'
+get_hospdata_wide <- function(hospdata_long,
+                              preferred_source = c("OWID", "ECDC")) {
+  preferred_source <- match.arg(preferred_source, several.ok = TRUE)
+
+  hospdata_wide <-
+    hospdata_long %>%
+    filter(source %in% preferred_source) %>%
+    select(-carry_fwd_value) %>%
+    tidyr::pivot_wider(names_from = indicator, values_from = value,
+                       names_glue = "{gsub(' ', '_',  tolower(indicator))}")
+
+  return(hospdata_wide)
+}
+
+#' Get latest hospitalization data by country, indicator, and data source
+#'
+#' @inheritParams get_hospdata_wide
+#'
+#' @return The same output as \code{\link{get_hospdata_long}}, except with one
+#'  row per country, indicator, and data source for the data from the most recent date
+#'  (so dates may vary by country, source, and indicator), and no column for `carry_fwd_value`.
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#'   hospdata_long <- get_hospdata_long()
+#'   hospdata_latest <- get_hospdata_latest(hospdata_long, preferred_source = "OWID")
+#'   head(hospdata_latest)
+#' }
+get_hospdata_latest <- function(hospdata_long,
```

Comment:
What are your thoughts on this API?

I started thinking about it more in response to wide, and wondered if there was any need for anything other than just a "long" version.

The testing accessors are kind of similar to this I guess, but `get_vax()`, `get_covid_df()`, etc. are all "long" by default. Is there code that relies on the `get_hospdata_latest()`-type of transformation already, or do you think we could just keep only `get_hospdata()` and let the user decide if they want only the latest data or not?

Reply:
I think just [here](https://github.com/CDCgov/ITF_Power_BI/blob/aebe03053d4814478075bee601e2c984f0836814/itf_dashboard/0_output_data.R#L258) and in the DUB it pulls in the file that's generated there. I think looking at the OG functions maybe the rationale was to avoid having to complete the data by date and just filtering to the most recent first. But I thought that would be harder to do in the pkg and since the longest part is just downloading all the data, didn't seem necessary to keep the download and combine separate from the complete step? 

How about I ditch get_hospdata_latest and get_hospdata_wide and instead add a couple of lines showing how to get those outputs to the example in get_hospdata_long? I can also just change that function then just to 'get_hospdata' :) 

Let me know!

Reply:
I like that idea.